src/Microsoft.Identity.Client/Internal/Cache/BaseTokenCacheItem.cs

```diff
@@ -36,78 +35,23 @@ namespace Microsoft.Identity.Client.Internal.Cache
     [DataContract]
     internal abstract class BaseTokenCacheItem
     {
-        protected RequestContext RequestContext;
-
         /// <summary>
         /// Default constructor.
         /// </summary>
-        internal BaseTokenCacheItem(string authority, string clientId, TokenResponse response)
+        public BaseTokenCacheItem(string clientId)
         {
-
-            if (response.IdToken!=null)
-            {
-                RawIdToken = response.IdToken;
-                IdToken idToken = IdToken.Parse(response.IdToken, RequestContext);
-                User = User.CreateFromIdToken(idToken);
-            }
-            
-            Authority = authority;
             ClientId = clientId;
         }
 
-        internal BaseTokenCacheItem()
+        public BaseTokenCacheItem()
         {
         }
 
-        internal IdToken IdToken { get; set; }
-
-        /// <summary>
-        /// Gets the ClientId.
-        /// </summary>
         [DataMember(Name = "client_id")]
         public string ClientId { get; set; }
```

Comment:
I can address this in future iterations. I do plan on doing an OOP sanity check across the board.